lib/contentTypeParser.js

```diff
@@ -67,11 +75,13 @@ ContentTypeParser.prototype.add = function (contentType, opts, parserFn) {
 }
 
 ContentTypeParser.prototype.hasParser = function (contentType) {
-  if (contentType === 'application/json') {
-    return this.customParsers['application/json'].fn !== this[kDefaultJsonParse]
-  }
-  if (contentType === 'text/plain') {
-    return this.customParsers['text/plain'].fn !== defaultPlainTextParser
+  return contentType in this.customParsers
+}
+
+ContentTypeParser.prototype.existingParser = function (contentType) {
+  this.nativeParsers['application/json'] = this[kDefaultJsonParse]
```

Comment:
Did not get why we need to assing here, could you explain?

Reply:
Well, it was the only way to get it to work. It took me quite some time to figure out why I could not get my solution to work and then this worked.

Looking at it now (away from my dev box now) I see its probably because of 

https://github.com/fastify/fastify/blob/master/lib/contentTypeParser.js#L240

If that was changed to:
```javascript
  contentTypeParser.nativeParsers['application/json'] = c.nativeParsers['application/json']
```
Then I think everything related to kDefaultJsonParse could be removed. 
However I can only test that once I'm back at my dev box in a week or so.

Reply:
thanks for explanation!

Reply:
FYI: I checked out of curiosity and yes that single line would have been enough ;-)